vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -150,14 +158,35 @@ public ISpace getFullyLoadedSpace(String id) {
      *            if id is null throws exception, else delete corresponding
      *            space
      * @throws SpaceDoesNotExistException 
+     * @throws SpaceLinkDoesNotExistException 
      */
     @Override
-    public void deleteSpaceById(String id) throws SpaceDoesNotExistException {
+    public void deleteSpaceById(String id) throws SpaceDoesNotExistException, SpaceLinkDoesNotExistException {
         try {
             spaceRepo.deleteById(id);
-        } catch (IllegalArgumentException | EmptyResultDataAccessException exception) {
-            throw new SpaceDoesNotExistException(exception);
+            linkManager.deleteSpaceLinkBySource(id);
+        } catch (IllegalArgumentException | EmptyResultDataAccessException | SpaceLinkDoesNotExistException exception) {
+            if(exception instanceof SpaceLinkDoesNotExistException)
+                throw new SpaceLinkDoesNotExistException(exception);
+            else
+                throw new SpaceDoesNotExistException(exception);
         }
 
     }
+    
+    /**
+     * Method to return identifiers of target spaces of all spacelinks
+     * 
+     * @return list of identifiers corresponding to target spaces in each spacelink 
+     */
+    @Override
+    public List<String> getAllTargetSpaceIds() {
```

Comment:
why don't you get all space links that have a specified space as target space? Such a method would probably much more reusable in the future.

Reply:
In spaces listing, returning all target spaces made more sense to me as I could use contains function in jsp to show warning message when trying to delete. If I am getting space links by target space, I might have to call get space links by target space for all spaces for spaces listing to do the same check. Let me know if I could do it differently.

Reply:
this could become a huge performance issue. There are potentially multiple space links pointing at a space (another issue here, since those would end up multiple times in your list), resulting in a potentially very big number of space links. Retrieving all of those every time you go to the space list, will make problems eventually.

Instead I would just check when the user actually clicks on the delete button, if there is a space link with that space as target and then show a warning. Even checking on page load for the 20 spaces (ideally the space list will be paginated if it is not yet), would most likely be less of a performance issue.